tests/Test/Proto/TH.hs

```diff
@@ -0,0 +1,120 @@
+{-# LANGUAGE QuasiQuotes #-}
+
+module Test.Proto.TH (testTree) where
+
+import Data.List.NonEmpty (NonEmpty (..))
+
+import Hedgehog (Property, property, (===), withTests)
+
+import Proto3.Suite.DotProto.AST 
+  ( DotProto (..)
+  , DotProtoDefinition (..)
+  , DotProtoIdentifier (..)
+  , DotProtoMessagePart (..)
+  , DotProtoMeta (..)
+  , DotProtoPackageSpec (..)
+  , DotProtoField (..)
+  , DotProtoPrimType (..)
+  , Path (..)
+  , DotProtoType (..)
+  )
+import Proto3.Suite.DotProto.Rendering ()
+
+import Test.Proto.Quote (dotProto3Syntax)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Hedgehog (testProperty)
+
+--------------------------------------------------------------------------------
+
+testTree :: TestTree
+testTree =
+  testGroup
+    "Test.Proto.TH"
+    [ testProperty "dotProtoQuoter" testDotProtoQuoter
+    ]
+
+testDotProtoQuoter :: Property
+testDotProtoQuoter = 
+  withTests 1 $ property do
+    protoDefinitions dotProtoParsed === protoDefinitions dotProtoExpected
+  where 
+    dotProtoExpected :: DotProto
+    dotProtoExpected = 
```

Comment:
Why bother with a full `DotProto`, instead of just defining the expectations for its `protoDefinitions` field?

Reply:
Checking the full `DotProto` would be tricky because the metadata field of `DotProto` tracks the protobuf package `Path`. The package used is the package identifier that gets exposed by `pkgName` [here in the definition of the quasi-quoter](https://github.com/awakesecurity/proto3-suite/blob/riz0id/proto3-quasiquote/tests/Test/Proto/Quote.hs#L26) by `thisModule`. The `pkgName` value then depends on if the file was built with `cabal new-repl` or `cabal v2-build`.

AFAIK there is no way to get around this beside not checking the meta field. So I really don't want perfect field-wise equality. I really only cared to check that the expression got lifted correctly. I can add additional assertions for the other fields that aren't the metadata fields.

Reply:
I see you added check of all but the metadata fields, so my comment no longer applies.  I'm fine with this as-is.